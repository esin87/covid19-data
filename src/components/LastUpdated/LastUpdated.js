import React from 'react';
import Moment from 'react-moment';

const LastUpdated = ({ date }) => {
	return (
		<p className='text-muted' style={{ marginTop: '0.5rem' }}>
			Last updated:{' '}
			<Moment fromNow ago>
				{date}
			</Moment>{' '}
			ago
		</p>
	);
};

export default LastUpdated;
