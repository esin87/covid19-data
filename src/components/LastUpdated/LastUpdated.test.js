import React from 'react';
import LastUpdated from './LastUpdated';
import { mount, render } from 'enzyme';

describe('LastUpdated component', () => {
	let component;
	const date = '2020-03-16T21:10:53.86852587Z';

	beforeEach(() => {
		component = mount(<LastUpdated date={date} />);
	});

	it('renders', () => {
		render(component);
	});

	it('should match the snapshot', () => {
		expect(component.html()).toMatchSnapshot();
	});

	it('contains a paragraph element with the appropriate text', () => {
		expect(component.find('p').text()).toBe('Last updated: a month ago');
	});

	it('receives the appropriate props', () => {
		expect(component.props()).toEqual({
			date: '2020-03-16T21:10:53.86852587Z',
		});
	});
});
