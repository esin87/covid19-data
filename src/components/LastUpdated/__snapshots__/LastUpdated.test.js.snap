// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LastUpdated component should match the snapshot 1`] = `"<p class=\\"text-muted\\" style=\\"margin-top: 0.5rem;\\">Last updated: <time datetime=\\"1584393053868\\">a month</time> ago</p>"`;
